### PR TITLE
feat: Query explanation

### PR DESCRIPTION
**Pull request by lionsardesai**
add query explanation UI

## Checklist

- [ ] I have run this code and it appears to resolve the stated issue
- [ ] `README.md` updated and added information about my change

### CODE REVIEW

File: src/webview_provider/queryResultPanel.ts [hunk 1]
```diff
@@ -85,10 +98,12 @@ export class QueryResultPanel implements WebviewViewProvider {
 
   private _disposables: Disposable[] = [];
   private _panel: WebviewView | undefined;
+  private adapter: string = "";
```
Comment (mdesmet) [inline src/webview_provider/queryResultPanel.ts:101]: empty string is not really clean. Should be rather undefined.

File: src/manifest/dbtProject.ts [hunk 2]
```diff
@@ -552,7 +552,7 @@ export class DBTProject implements Disposable {
           "Encountered an unknown issue: " + exc + ".",
         ),
       );
-      return "Detailed error information:\n" + exc;
+      return "Exception:\n\nDetailed error information:\n" + exc;
```
Comment (mdesmet) [inline src/manifest/dbtProject.ts:555]: Why this change?

File: src/commands/runModel.ts [hunk 3]
```diff
@@ -53,7 +53,7 @@ export class RunModel {
     }
   }
 
-  executeQueryOnActiveWindow() {
+  _getQueryWithName() {
```
Comment (mdesmet) [inline src/commands/runModel.ts:56]: why not use a private method?

Comment (lionsardesai) [reply src/commands/runModel.ts:56]: done.. python habits.

File: src/commands/runModel.ts [hunk 4]
```diff
@@ -67,7 +67,25 @@ export class RunModel {
...
+    const { query, queryName } = ret;
+    this.executeSQL(window.activeTextEditor!.document.uri, query, queryName);
+  }
+
+  getSummaryOnActiveWindow() {
+    const ret = this._getQueryWithName();
+    if (ret === undefined) {
+      return;
+    }
+    const { query, queryName } = ret;
```
[the remark this answers is not in the document]
Comment (lionsardesai) [reply src/commands/runModel.ts:87]: removed

File: src/webview_provider/queryResultPanel.ts (hunk 1 above)
Comment (lionsardesai) [reply src/webview_provider/queryResultPanel.ts:101]: setting it to unknown to match dbtProject.ts

File: src/manifest/dbtProject.ts (hunk 2 above)
Comment (lionsardesai) [reply src/manifest/dbtProject.ts:555]: Adding this because this function still returns a string after catching an exception. All exceptions have Exception: in the beginning so matching those. if string starts with that, i'm not generating summary for it.

Comment (lionsardesai) [reply src/manifest/dbtProject.ts]: returning undefined if exception encountered after discussion with michiel.

**Review by mdesmet — APPROVED** [review]
